### PR TITLE
Update 0.19.0 rc1

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

File: src/main/scala/dotty/xml/interpolator/package.scala [hunk 1]
```diff
@@ -5,5 +5,5 @@ import scala.quoted._
 type Scope = scala.xml.NamespaceBinding
 implicit val top: Scope = scala.xml.TopScope
 
-inline def (ctx: => StringContext) xml (args: => (given Scope => Any)*) given Scope <: Any =
```
Comment (anatoliykmetyuk) [inline src/main/scala/dotty/xml/interpolator/package.scala]: `Scope <: Any` is no longer permitted – @nicolasstucki can we just drop it?

Comment (smarter) [reply src/main/scala/dotty/xml/interpolator/package.scala]: The `<: Any` part is the result type of the inline method, so it should now be written `inline def ...(given Scope) <: Any = `

Comment (anatoliykmetyuk) [reply src/main/scala/dotty/xml/interpolator/package.scala]: Oh. Well... Good thing we now need parentheses around givens to avoid such a confusion.

Comment (nicolasstucki) [reply src/main/scala/dotty/xml/interpolator/package.scala]: Clearly the old syntax was confusing

File: src/main/scala/dotty/xml/interpolator/package.scala [hunk 2]
```diff
@@ -5,5 +5,5 @@ import scala.quoted._
...
-  ${ dotty.xml.interpolator.internal.Macro.impl('ctx, 'args, '{the[Scope]}) }
\ No newline at end of file
+inline def (ctx: => StringContext) xml (args: => ((given Scope) => Any)*)(given Scope) =
```
Comment (nicolasstucki) [inline src/main/scala/dotty/xml/interpolator/package.scala]:
```suggestion
inline def (ctx: => StringContext) xml (args: => ((given Scope) => Any)*)(given Scope) <: Any =
```

**Comment by anatoliykmetyuk** [comment]
@nicolasstucki I've addressed all the "mechanical" sytnax corrections, only three errors are now left:

<details><summary>Errors</summary>

```scala
[info] Compiling 6 Scala sources to /Users/anatolii/Projects/dotty/tools/dotty-release/ecosystem_projects/xml-interpolator/target/scala-0.19/test-classes ...
[error] -- Error: /Users/anatolii/Projects/dotty/tools/dotty-release/ecosystem_projects/xml-interpolator/src/test/scala/dotty/xml/interpolator/InterpolationTest.scala:43:26
[error] 43 |    assert(xml"""<foo a=${xml"<bar/><baz/>"}/>""" ≈ <foo a={<bar/><baz/>}/>)
[error]    |                          ^^^^^^^^^^^^^^^^^
[error]    |type mismatch;
[error]    | found   : scala.xml.NodeBuffer
[error]    | required: scala.Predef.String | scala.collection.immutable.Seq[scala.xml.Node] | scala.Option[scala.collection.immutable.Seq[scala.xml.Node]]
[error]    |
[error]    | This location is in code that was inlined at InterpolationTest.scala:43
[error] -- Error: /Users/anatolii/Projects/dotty/tools/dotty-release/ecosystem_projects/xml-interpolator/src/test/scala/dotty/xml/interpolator/ReporterTest.scala:124:17
[error] 124 |    assertEquals(xml"<foo bar='' bar=''/>", List((12, "attribute `bar` may only be defined once")))
[error]     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^
[error]     |Exception occurred while executing macro expansion.
[error]     |scala.tasty.reflect.ExprCastError: Expr: /* inlined from dotty.xml.interpolator.internal.Expand$ */
[error]     |  {
[error]     |    new _root_.scala.xml.NodeBuffer()
[error]     |  }
[error]     |did not conform to type: Option[Seq[xml.Node]]
[error]     |
[error]     |	at dotty.tools.dotc.tastyreflect.ReflectionCompilerInterface.QuotedExpr_cast(ReflectionCompilerInterface.scala:1740)
[error]     |	at dotty.tools.dotc.tastyreflect.ReflectionCompilerInterface.QuotedExpr_cast(ReflectionCompilerInterface.scala:1732)
[error]     |	at scala.tasty.reflect.QuotedOps$QuotedExprAPI.cast(QuotedOps.scala:13)
[error]     |	at dotty.xml.interpolator.internal.Expand$.expandAttributes$$anonfun$1(Expand.scala:83)
[error]     |	at scala.collection.immutable.List.foldRight(List.scala:324)
[error]     |	at dotty.xml.interpolator.internal.Expand$.expandAttributes(Expand.scala:87)
[error]     |	at dotty.xml.interpolator.internal.Expand$.expandElem(Expand.scala:41)
[error]     |	at dotty.xml.interpolator.internal.Expand$.expandNode(Expand.scala:19)
[error]     |	at dotty.xml.interpolator.internal.Expand$.apply(Expand.scala:12)
[error]     |	at dotty.xml.interpolator.internal.Macro$.$anonfun$5(Macro.scala:74)
[error]     |	at scala.Function1.$anonfun$andThen$1(Function1.scala:85)
[error]     |	at dotty.xml.interpolator.internal.Macro$.implCore(Macro.scala:77)
[error]     |	at dotty.xml.interpolator.internal.Macro$.implErrors(Macro.scala:56)
[error]     |
[error]     | This location is in code that was inlined at ReporterTest.scala:124
[error] -- Error: /Users/anatolii/Projects/dotty/tools/dotty-release/ecosystem_projects/xml-interpolator/src/test/scala/dotty/xml/interpolator/SimpleNodeTest.scala:85:11
[error] 85 |    assert(xml"""<foo a="Hello &name;!"/>""" ≈ <foo a="Hello &name;!"/>)
[error]    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |Exception occurred while executing macro expansion.
[error]    |scala.tasty.reflect.ExprCastError: Expr: /* inlined from dotty.xml.interpolator.internal.Expand$ */
[error]    |  {
[error]    |    /* inlined from outside */
[error]    |      {
[error]    |        /* inlined from dotty.xml.interpolator.internal.Expand$ */
[error]    |          {
[error]    |            /* inlined from outside */
[error]    |              {
[error]    |                /* inlined from dotty.xml.interpolator.internal.Expand$ */
[error]    |                  {
[error]    |                    /* inlined from outside */
[error]    |                      {
[error]    |                        /* inlined from dotty.xml.interpolator.internal.Expand$
[error]    |                           */
[error]    |                        {
[error]    |                          new _root_.scala.xml.NodeBuffer()
[error]    |                        }
[error]    |                      }
[error]    |                    .&+(
[error]    |                      /* inlined from outside */
[error]    |                        {
[error]    |                          /* inlined from
[error]    |                            dotty.xml.interpolator.internal.Expand$
[error]    |                           */
[error]    |                            {
[error]    |                              new _root_.scala.xml.Text(
[error]    |                                /* inlined from outside */
[error]    |                                  {
[error]    |                                    "Hello "
[error]    |                                  }
[error]    |                              )
[error]    |                            }
[error]    |                        }
[error]    |                    )
[error]    |                  }
[error]    |              }
[error]    |            .&+(
[error]    |              /* inlined from outside */
[error]    |                {
[error]    |                  /* inlined from dotty.xml.interpolator.internal.Expand$ */
[error]    |                    {
[error]    |                      new _root_.scala.xml.EntityRef(
[error]    |                        /* inlined from outside */
[error]    |                          {
[error]    |                            "name"
[error]    |                          }
[error]    |                      )
[error]    |                    }
[error]    |                }
[error]    |            )
[error]    |          }
[error]    |      }
[error]    |    .&+(
[error]    |      /* inlined from outside */
[error]    |        {
[error]    |          /* inlined from dotty.xml.interpolator.internal.Expand$ */
[error]    |            {
[error]    |              new _root_.scala.xml.Text(
[error]    |                /* inlined from outside */
[error]    |                  {
[error]    |                    "!"
[error]    |                  }
[error]    |              )
[error]    |            }
[error]    |        }
[error]    |    )
[error]    |  }
[error]    |did not conform to type: Option[Seq[xml.Node]]
[error]    |
[error]    |	at dotty.tools.dotc.tastyreflect.ReflectionCompilerInterface.QuotedExpr_cast(ReflectionCompilerInterface.scala:1740)
[error]    |	at dotty.tools.dotc.tastyreflect.ReflectionCompilerInterface.QuotedExpr_cast(ReflectionCompilerInterface.scala:1732)
[error]    |	at scala.tasty.reflect.QuotedOps$QuotedExprAPI.cast(QuotedOps.scala:13)
[error]    |	at dotty.xml.interpolator.internal.Expand$.expandAttributes$$anonfun$1(Expand.scala:83)
[error]    |	at scala.collection.immutable.List.foldRight(List.scala:324)
[error]    |	at dotty.xml.interpolator.internal.Expand$.expandAttributes(Expand.scala:87)
[error]    |	at dotty.xml.interpolator.internal.Expand$.expandElem(Expand.scala:41)
[error]    |	at dotty.xml.interpolator.internal.Expand$.expandNode(Expand.scala:19)
[error]    |	at dotty.xml.interpolator.internal.Expand$.apply(Expand.scala:12)
[error]    |	at dotty.xml.interpolator.internal.Macro$.$anonfun$5(Macro.scala:74)
[error]    |	at scala.Function1.$anonfun$andThen$1(Function1.scala:85)
[error]    |	at dotty.xml.interpolator.internal.Macro$.implCore(Macro.scala:77)
[error]    |	at dotty.xml.interpolator.internal.Macro$.impl(Macro.scala:32)
[error]    |
[error]    | This location is in code that was inlined at SimpleNodeTest.scala:85
[error] three errors found
```
</details>

It looks like `NodeBuffer` does not qualify as an immutable sequence.

**Comment by bishabosha** [comment]
closing because the 0.20.0-RC1 release was merged

**Pull request closed without merging**